IHP/Mail.hs

```diff
@@ -29,8 +31,8 @@ buildMail mail = let ?mail = mail in simpleMail (to mail) from subject (cs $ tex
 -- | Sends an email
 --
 -- Uses the mail server provided in the controller context, configured in Config/Config.hs
-sendMail :: (Config.FrameworkConfig, BuildMail mail) => mail -> IO ()
-sendMail mail = buildMail mail >>= sendWithMailServer Config.mailServer
+sendMail :: (BuildMail mail, ?requestContext :: RequestContext ) => mail -> IO ()
```

Comment:
I'm a bit worried about this change. When using e.g. a Script (or generally anything running outside the request-response cycle, such as a worker queue) this will not be callable as the `?requestContext` is of course not available.

The only solution that came to my mind was to have some kind of stub request context for Scripts and the job queue.

Reply:
This is indeed one of the most blatant assumptions I made. I was developing on a fresh project so I can't foresee if this breaks anything. Would be nice to have an all out project that uses all features, that makes developing easier since things break more often.

Reply:
The example given in the documentation only shows using this function from within a controller action at which point you do have access to the RequestContext. Or did you have plans of providing a more generic use case?

Reply:
I think I found a solution by requiring only a FrameworkConfig that gets injected when setting up the GHCI environment

Reply:
Yes, we have plans to add a email queue, so that the emails would be send in the background (this makes e.g. actions that send emails way faster). For that we just need to merge our internal job queue into IHP. The job runner is build using an IHP script.

For now the `sendMailFromScript` solution looks good :)